brax/training/spectral_norm.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2021 The Brax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Flax-style Dense module with Spectral Normalization.
+
+Reference:
+  Dense: https://github.com/google/flax/blob/main/flax/linen/linear.py
+  Spectral Normalization:
+    - https://arxiv.org/abs/1802.05957
+    - https://github.com/deepmind/dm-haiku/blob/main/haiku/_src/spectral_norm.py
+"""
+from typing import (Any, Callable, Tuple)
+
+from jax import lax
+import jax.numpy as jnp
+
+from flax import linen
+from flax.linen.initializers import lecun_normal, normal, zeros
+
+
+default_kernel_init = lecun_normal()
```

Comment:
style nit: module level constants should be all-caps: `DEFAULT_KERNEL_INIT`

that said, since you are only using this in one place, I think the code is a bit more readable if you just do this below:

```
kernel_init: Callable[[PRNGKey, Shape, Dtype], Array] = lecun_normal()
```